idp-service/src/test/java/io/mosip/idp/AuthorizationCodeFlowParameterizedTest.java

```diff
@@ -0,0 +1,315 @@
+import com.nimbusds.jwt.JWTClaimsSet;
+import com.nimbusds.jwt.SignedJWT;
+import io.mosip.idp.core.dto.*;
+import io.mosip.idp.core.spi.AuthenticationWrapper;
+import io.mosip.idp.core.spi.TokenService;
+import io.mosip.idp.core.util.AuthenticationContextClassRefUtil;
+import io.mosip.idp.core.util.Constants;
+import io.mosip.idp.repository.ClientDetailRepository;
+import io.mosip.idp.services.CacheUtilService;
+import lombok.extern.slf4j.Slf4j;
+import org.junit.*;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.TestContextManager;
+import org.springframework.test.context.junit4.rules.SpringClassRule;
+import org.springframework.test.context.junit4.rules.SpringMethodRule;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+
+import java.time.Instant;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Date;
+import java.util.HashMap;
+
+import static io.mosip.idp.core.util.Constants.UTC_DATETIME_PATTERN;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@RunWith(Parameterized.class)
+@SpringBootTest
+@AutoConfigureMockMvc(secure = false)
+@Slf4j
+public class AuthorizationCodeFlowParameterizedTest {
+
+    @ClassRule
+    public static final SpringClassRule SPRING_CLASS_RULE = new SpringClassRule();
+
+    @Rule
+    public final SpringMethodRule springMethodRule = new SpringMethodRule();
+
+    @Autowired
+    MockMvc mockMvc;
+
+    @Autowired
+    ObjectMapper objectMapper;
+
+    @Autowired
+    private ClientDetailRepository clientDetailRepository;
+
+    @Autowired
+    private AuthenticationWrapper authenticationWrapper;
+
+    @Autowired
+    private CacheUtilService cacheUtilService;
+
+    @Autowired
+    private TokenService tokenService;
+
+    @Autowired
+    private AuthenticationContextClassRefUtil authenticationContextClassRefUtil;
+
+    private TestContextManager testContextManager;
+
+    private String testName;
+    private String clientId;
+    private String redirectionUrl;
+    private String state;
+    private String nonce;
+    private static JWK clientJWK = TestUtil.generateJWK_RSA();
+
+    public AuthorizationCodeFlowParameterizedTest(String testName, String clientId, String redirectUrl, String state, String nonce) {
```

Comment:
This covers positive full flow, but various behavious other behavious we have covered else where ?

Reply:
We are still working on the test cases, I have committed basic test cases for now.